src/category_theory/monoidal/category.lean

```diff
@@ -0,0 +1,405 @@
+-- tensor unit:
+(tensor_unit              : C)
+-- associator:
+(associator               :
+  Π X Y Z : C, (tensor_obj (tensor_obj X Y) Z) ≅ (tensor_obj X (tensor_obj Y Z)))
+(associator_naturality'   :
+  assoc_natural tensor_obj @tensor_hom associator . obviously)
+-- left unitor:
+(left_unitor              : Π X : C, tensor_obj tensor_unit X ≅ X)
+(left_unitor_naturality'  :
+  left_unitor_natural tensor_obj @tensor_hom tensor_unit left_unitor . obviously)
+-- right unitor:
+(right_unitor             : Π X : C, tensor_obj X tensor_unit ≅ X)
+(right_unitor_naturality' :
+  right_unitor_natural tensor_obj @tensor_hom tensor_unit right_unitor . obviously)
+-- pentagon identity:
+(pentagon'                : pentagon @tensor_hom associator . obviously)
+-- triangle identity:
+(triangle'                :
+  triangle @tensor_hom left_unitor right_unitor associator . obviously)
+
+restate_axiom monoidal_category.tensor_map_id'
+attribute [simp] monoidal_category.tensor_map_id
+restate_axiom monoidal_category.tensor_map_comp'
+attribute [simp] monoidal_category.tensor_map_comp
+restate_axiom monoidal_category.associator_naturality'
+restate_axiom monoidal_category.left_unitor_naturality'
+restate_axiom monoidal_category.right_unitor_naturality'
+restate_axiom monoidal_category.pentagon'
+restate_axiom monoidal_category.triangle'
+attribute [simp] monoidal_category.triangle
+
+open monoidal_category
+
+infixr ` ⊗ `:70 := tensor_obj
+infixr ` ⊗ `:70 := tensor_hom
+
+def tensor_iso {C : Sort u} {X Y X' Y' : C} [monoidal_category.{v} C] (f : X ≅ Y) (g : X' ≅ Y') :
+    X ⊗ X' ≅ Y ⊗ Y' :=
+{ hom := f.hom ⊗ g.hom,
+  inv := f.inv ⊗ g.inv,
+  hom_inv_id' := by rw [←tensor_map_comp, iso.hom_inv_id, iso.hom_inv_id, ←tensor_map_id],
+  inv_hom_id' := by rw [←tensor_map_comp, iso.inv_hom_id, iso.inv_hom_id, ←tensor_map_id] }
+
+infixr ` ⊗ `:70 := tensor_iso
+
+section
+
+variables {C : Sort u} [𝒞 : monoidal_category.{v} C]
+include 𝒞
+
+instance tensor_is_iso {W X Y Z : C} (f : W ⟶ X) [is_iso f] (g : Y ⟶ Z) [is_iso g] : is_iso (f ⊗ g) :=
+{ ..(as_iso f ⊗ as_iso g) }
+
+@[simp] lemma inv_tensor {W X Y Z : C} (f : W ⟶ X) [is_iso f] (g : Y ⟶ Z) [is_iso g] :
+  inv (f ⊗ g) = inv f ⊗ inv g
+:= rfl
+
+variables {U V W X Y Z : C}
+
+-- When `rewrite_search` lands, add @[search] attributes to
+
+-- monoidal_category.tensor_map_id monoidal_category.tensor_map_comp monoidal_category.associator_naturality
+-- monoidal_category.left_unitor_naturality monoidal_category.right_unitor_naturality
+-- monoidal_category.pentagon monoidal_category.triangle
+
+-- interchange interchange_left_identity interchange_right_identity interchange_identities
+-- triangle_1 triangle_2 triangle_3 triangle_4
+-- left_unitor_product left_unitor_product_inv
+-- right_unitor_product right_unitor_product_inv
+-- pentagon_inv
+-- associator_inv_naturality
+-- left_unitor_inv_naturality
+-- right_unitor_inv_naturality
+
+lemma interchange (f : U ⟶ V) (g : V ⟶ W) (h : X ⟶ Y) (k : Y ⟶ Z)
```

Comment:
I would prefer to see this name follow the naming convention... I proposed upstairs that we rename `tensor_map_comp` to `tensor_comp`. We could the same here...

Reply:
Removed `interchange` entirely, and renamed `interchange_*` variants.